inc/math/linalg.hpp

```diff
@@ -379,20 +381,100 @@ namespace exo
 				return r;
 			}
 
-			template <ssize_t MC>
-			Vec<S, R> operator* (Vec<S, MC>& v)
+
+			Vec<S, R> operator* (Vec<S, C>& v) { return *this * std::move(v); }
+			Vec<S, R> operator* (Vec<S, C>&& v)
 			{
 				Vec<S, R> r;
 
 				for (int row = R; row--;)
-				for (int col = MC; col--;)
+				for (int col = C; col--;)
 				{
 					r[row] += this->m[row][col] * v[col];
 				}
 
 				return r;
 			}
 
+			Mat<S, R, C> operator* (S s)
+			{
+				Mat<S, R, C> r;
+
+				for (int row = R; row--;)
+				for (int col = C; col--;)
+				{
+					r[row][col] = this->m[row][col] * s;
+				}
+
+				return r;
+			}
+
+			Mat<S, R, C>& operator*= (S s)
+			{
+				for (int row = R; row--;)
+				for (int col = C; col--;)
+				{
+					m[row][col] *= s;
+				}
+
+				return *this;
+			}
+
+			Mat<S, R, C> operator/ (S s)
+			{
+				Mat<S, R, C> r;
+
+				for (int row = R; row--;)
+				for (int col = C; col--;)
+				{
+					r[row][col] = this->m[row][col] / s;
+				}
+
+				return r;
+			}
+
+			Mat<S, R, C>& operator/= (S s)
+			{
+				Mat<S, R, C> r;
+
+				for (int row = R; row--;)
+				for (int col = C; col--;)
+				{
+					m[row][col] /= s;
+				}
+
+				return *this;
+			}
+
+			Mat<S, R, C> operator+ (Mat<S, R, C>& m) { return *this + std::move(m); }
+			Mat<S, R, C> operator+ (Mat<S, R, C>&& m)
```

Comment:
If you want to move `m`, why even create `r`? Couldn't you just do `m[row][col] += this->m[row][col];`?

Reply:
Are you suggesting something like `Mat<S, R, C> r(*this);` then doing what you described above? This overload is intended to return a copy of the resulting addition, not mutate the existing matrix.

Reply:
I am suggesting:
```
Mat<S, R, C> operator+ (Mat<S, R, C>&& m)
{
	for (int row = R; row--;)
	for (int col = C; col--;)
	{
		m[row][col] += this->m[row][col]];
	}
	return m;
}
```
I did a little testing and now I'm not so sure, though. If your tests pass, I'd say go for it.

Reply:
Ah, so a few things here. `Mat` objects don't contain pointers. They are static 2d arrays that are generated at compile time thanks to the magic of templates. So you literally can't swap the `m`s like you show in that example. Plus, even if you could, this would set the operand's `m` to nullptr which would be bad.

Reply:
if they were std::arrays, you could call std::swap on them.

Reply:
You could also implement a move constructor that does the swap.